src/squlearn/kernel/matrix/kernel_optimizer.py

```diff
@@ -0,0 +1,89 @@
+"""Quantum Kernel Optimizer"""
+
+import numpy as np
+from functools import partial
+from typing import Optional, Sequence
+
+from squlearn.kernel.matrix.kernel_matrix_base import KernelMatrixBase
+from squlearn.kernel.optimization.kernel_loss_base import KernelLossBase
+from squlearn.optimizers.optimizer_base import OptimizerBase
+
+
+class KernelOptimizer(KernelMatrixBase):
+    """
+    Quantum kernel optimizer.
+    This class can be used to optimize the variational parameters of a quantum kernel.
+
+    Args:
+        loss (KernelLossBase): The loss function to be minimized.
+        optimizer (OptimizerBase): The optimizer to be used.
+        quantum_kernel (KernelMatrixBase): The quantum kernel to be optimized.
+        initial_parameters (Optional[Sequence[float]]): Initial parameters for the optimizer.
+    """
+
+    def __init__(
+        self,
+        quantum_kernel: KernelMatrixBase,
+        loss: KernelLossBase = None,
+        optimizer: OptimizerBase = None,
+        initial_parameters: Optional[Sequence[float]] = None,
+    ) -> None:
+        super().__init__(
+            encoding_circuit=quantum_kernel.encoding_circuit,
+            executor=quantum_kernel._executor,
+            initial_parameters=initial_parameters,
+            parameter_seed=quantum_kernel._parameter_seed,
+            regularization=quantum_kernel._regularization,
+        )
+        self._quantum_kernel = quantum_kernel
+        self._loss = loss
+        self._optimizer = optimizer
+        self._initial_parameters = (
+            initial_parameters
+            if initial_parameters is not None
+            else self._quantum_kernel.parameters
+        )
+        self._opt_result = None
+        self._optimal_value = None
+        self._optimal_point = None
+        self._is_trainable = True
+
+    def run_optimization(self, X: np.ndarray, y: np.ndarray = None):
+        """Run the optimization and return the result.
+
+        Args:
+            X (np.ndarray): The input data.
+            y (np.ndarray): The labels.
+
+        Returns:
+            OptimizeResult: The optimization result.
+        """
+        num_params = self._quantum_kernel.num_parameters
+        if num_params == 0:
+            raise ValueError(
+                "Quantum kernel cannot be fit because there are no training parameters specified."
+            )
+
+        # Perform kernel optimization
+        loss_function = partial(self._loss.compute, data=X, labels=y)
+        opt_result = self._optimizer.minimize(fun=loss_function, x0=self._initial_parameters)
+        self._optimal_value = opt_result.fun
+        self._optimal_point = opt_result.x
+        self._opt_result = opt_result
+
+        # Assign optimal parameters to the quantum kernel
+        self.assign_parameters(self._opt_result.x)
```

Comment:
Is this working? Somewhere I'm missing the call to ``self._quantum_kernel.assign_parameters()``